xbmc/addons/AddonManager.cpp

```diff
@@ -610,6 +614,10 @@ bool CAddonMgr::FindAddon(const std::string& addonId, const AddonVersion& addonV
 
   m_installedAddons[addonId] = it->second; // insert/replace entry
 
+  // Sync with db
+  installed.insert(it->second->ID());
+  m_database.SyncInstalled(installed, m_systemAddons, m_optionalSystemAddons);
```

Comment:
Wrong. It is not guaranteed that `installed` actually contains what is installed.

The fix must be to only put `it->second`  into the database as newly installed.

Reply:
ok, i suspected it would reintroduce the race.
then i'll see how to do it correctly ;)